### PR TITLE
Connect gift card update to api

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it connects gift card update page to api and adds gift card update balance logic

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://feature-gift-cards.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3912 Add Gift Card update view header, basic details & metadata fields](https://app.clickup.com/t/2549495/SALEOR-3912)

File: src/giftCards/GiftCardUpdatePage/GiftCardUpdateInfoCard/GiftCardUpdateInfoCardContent.tsx [hunk 1]
```diff
@@ -47,6 +46,18 @@ const GiftCardUpdateInfoCard: React.FC = () => {
     name: string;
     url?: string;
   } => {
+    // createdBy can be either customer or staff hence
+    // we check for issued event
+    if (!!cardIssuedEvent) {
```
Comment (jwm0) [inline src/giftCards/GiftCardUpdatePage/GiftCardUpdateInfoCard/GiftCardUpdateInfoCardContent.tsx:51]:
```suggestion
    if (cardIssuedEvent) {
```

File: src/giftCards/GiftCardUpdatePage/providers/GiftCardUpdateFormProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    });
+  };
+
+  const formErrors = getFormErrors(
+    ["tag", "expiryDate", "expiryPeriod"],
+    updateGiftCardOpts?.data?.giftCardUpdate?.errors
+  );
+
+  return (
+    <Form initial={getInitialData()} onSubmit={handleSubmit}>
```
Comment (jwm0) [inline src/giftCards/GiftCardUpdatePage/providers/GiftCardUpdateFormProvider.tsx:105]: This component will be rerendered quite often, see if there are any noticeable performance issues. In general all providers that do not immediately return `children` suffer from a number of rerenders. A quick fix is to `memo` the `Form` component and memoize the props to keep static references. `getInitialData` could be a useMemo and `handleSubmit` could use useCallback. See how often `Form` rerenders here by placing a `console.log` and count how many times it's called.

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/giftCards/GiftCardUpdatePage/providers/GiftCardUpdateFormProvider.tsx (hunk 2 above)
Comment (mmarkusik) [reply src/giftCards/GiftCardUpdatePage/providers/GiftCardUpdateFormProvider.tsx:105]: So far nothing seems to be out of order, going to keep an eye on this form in the incoming pr's.

**Review by SektorDV — APPROVED** [review]

**Pull request merged**